### ISSUE TITLE
Where is the GPU setup?  Running shows that GPU = None, in fact, I have a gpu, cuda10 installation training is no problem

### ISSUE THREAD

**Issue by QQ2737499951** [princeton-vl/CornerNet-Lite#48]
Excuse me?
Where is the GPU setup?

Running shows that GPU = None, in fact, I have a gpu, cuda10 installation training is no problem. thank you.

请问一下，gpu在哪里设置？运行显示 gpu=None，实际我gpu是有的，cuda10都安装训练没问题。方便加QQ沟通一下吗？ QQ2737499951,谢谢

================
args= Namespace(cfg_file='CornerNet_Saccade', dist_backend='nccl', dist_url=None, distributed=False, gpu=None, initialize=False, rank=0, start_iter=0, workers=2, world_size=-1)
train tart_iter 0 distributed False world_size -1 initialize False gpu None
Process 0: 创建模型 building model...
total parameters: 116969339
启动预取数据 start prefetching data...

**Comment by heilaw** [comment]
Can you include the complete error message?

**Comment by QQ2737499951** [comment]
> Can you include the complete error message?



D:\pytorch\CornerNet-Lite>python train.py CornerNet
ngpus_per_node= 1
准备读取参数 cfg_file= ./configs\CornerNet.json
Process 0: 加载所有数据集 loading all datasets...
Process 0: using 1 workers
coco_dir = B:/COCO2017/data\coco
self._split A train2017
self._data_dir B B:/COCO2017/data\coco\images\train2017
_anno_file C B:/COCO2017/data\coco\annotations\instances_train2017.json
loading annotations into memory...
Done (t=13.10s)
creating index...
index created!
coco_dir = B:/COCO2017/data\coco
self._split A val2017
self._data_dir B B:/COCO2017/data\coco\images\val2017
_anno_file C B:/COCO2017/data\coco\annotations\instances_val2017.json
loading annotations into memory...
Done (t=1.13s)
creating index...
index created!
系统配置 system config...
{'batch_size': 1,
 'cache_dir': './cache',
 'chunk_sizes': [1],
 'config_dir': './config',
 'data_dir': 'B:/COCO2017/data',
 'data_rng': <mtrand.RandomState object at 0x000001B2AA7ABCA8>,
 'dataset': 'COCO',
 'decay_rate': 10,
 'display': 100,
 'learning_rate': 0.00025,
 'max_iter': 500000,
 'nnet_rng': <mtrand.RandomState object at 0x000001B2AA7ABCF0>,
 'opt_algo': 'adam',
 'prefetch_size': 5,
 'pretrain': None,
 'result_dir': './results',
 'sampling_function': 'cornernet',
 'snapshot': 5000,
 'snapshot_name': 'CornerNet',
 'stepsize': 450000,
 'test_split': 'testing',
 'train_split': 'training',
 'val_iter': 100,
 'val_split': 'validation'}
数据库配置 db config...
{'ae_threshold': 0.5,
 'att_max_crops': 8,
 'att_nms_ks': [3, 3, 3],
 'att_ranges': [[96, 256], [32, 96], [0, 32]],
 'att_ratios': [16, 8, 4],
 'att_scales': [1, 1.5, 2],
 'att_sizes': [[16, 16], [32, 32], [64, 64]],
 'att_thresholds': [0.3, 0.3, 0.3, 0.3],
 'border': 128,
 'categories': 80,
 'data_aug': True,
 'gaussian_bump': True,
 'gaussian_iou': 0.3,
 'gaussian_radius': -1,
 'init_sizes': [192, 255],
 'input_size': [511, 511],
 'lighting': True,
 'max_per_image': 100,
 'max_per_set': 40,
 'max_scale': 32,
 'merge_bbox': False,
 'min_scale': 16,
 'nms_algorithm': 'exp_soft_nms',
 'nms_kernel': 3,
 'nms_threshold': 0.5,
 'num_dets': 1000,
 'output_sizes': [[128, 128]],
 'rand_center': True,
 'rand_color': True,
 'rand_crop': True,
 'rand_scale_max': 1.4,
 'rand_scale_min': 0.6,
 'rand_scale_step': 0.1,
 'rand_scales': array([0.6, 0.7, 0.8, 0.9, 1. , 1.1, 1.2, 1.3]),
 'ref_dets': True,
 'score_threshold': 0.05,
 'test_flipped': True,
 'test_scales': [1],
 'top_k': 100,
 'view_sizes': [],
 'weight_exp': 8}
len of db: 118287
分布式 distributed: False
**正式train args= Namespace(cfg_file='CornerNet', dist_backend='nccl', dist_url=None, distributed=False, gpu=None, initialize=False, rank=0, start_iter=0, workers=1, world_size=-1)**
**train..... tart_iter 0 distributed False world_size -1 initialize False gpu None
![BBBBBBBB](https://user-images.githubusercontent.com/33048209/57293373-1c90de80-70f7-11e9-8980-b287cad11d1b.png)
**
Process 0: 创建模型 building model...
total parameters: 201035212
system_config= <core.config.SystemConfig object at 0x00000277868FB240> db= <core.dbs.coco.COCO object at 0x00000277868FB7B8> queue= <multiprocessing.queues.Queue object at 0x0000027843333710> sample_data= <function data_sampling_func at 0x00000277862A60D0> data_aug= True
启动预取数据 start prefetching data...
洗牌指数 shuffling indices...
system_config= <core.config.SystemConfig object at 0x000001D6A507B240> db= <core.dbs.coco.COCO object at 0x000001D6A507B7B8> queue= <multiprocessing.queues.Queue object at 0x000001D6ACC60F28> sample_data= <function data_sampling_func at 0x000001D6A4A560D0> data_aug= False
启动预取数据 start prefetching data...
洗牌指数 shuffling indices...
setting learning rate to: 0.00025
开始训练 training start...
Process 0: 迭代 100: 训练损失 11.960190773010254
Process 0: 迭代 100: 验证损失 8.793777465820312
Process 0: 迭代 200: 训练损失 7.868043422698975
Process 0: 迭代 200: 验证损失 10.156821250915527
  0%|                                  | 149/500000 [03:04<172:18:18,  1.24s/it]


=============
D:\pytorch\CornerNet-Lite\train.py
259        pprint.pprint(training_dbs[0].configs)

261        print("len of db: {}".format(len(training_dbs[0].db_inds)))
262        print("分布式 distributed: {}".format(args.distributed))

264    **print("正式train args=",args)**
265    train(training_dbs, validation_db, system_config, model, args)
